### DISCUSSION TITLE
What if I want to modify the traceflags of the spancontext in the span? Is it currently supported?

### DISCUSSION THREAD

**Discussion by wuboui** [open-telemetry/opentelemetry-go#1765] [Q&A] [answered]
What if I want to modify the traceflags of the spancontext in the span? Is it currently supported?

**Comment by MrAlias** [comment, marked as answer]
You can use [`WithTraceFlags`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext.WithTraceFlags) to set the trace flags for a copy of a `SpanContext`.

A `SpanContext` is an immutable type by design and it's association with a span is also not mutable. If you are asking how to change this association, the answer is the same as the other question you asked [here](https://github.com/open-telemetry/opentelemetry-go/discussions/1760): you cannot by design.
